### PR TITLE
Layer component

### PR DESCRIPTION
**Pull request by chrislloyd**
This is the start of making Flyout/Modal more sane (and easy to use). This component creates a "layer" - basically a portal to the end of the DOM tree - so anything inside of it is guaranteed to be at the top of any zIndex rules. It uses React 16 portals so updates/events flow between each nicely.

There are no tests as `react-test-renderer` doesn't support Portals. However, I've manually tested with Modal.

### CODE REVIEW

**Comment by christianvuerings** [comment]
Deploy preview for *gestalt* ready!

Built with commit 94c200c10ab1e15fad7826b10d1c2a48c39b4249

https://deploy-preview-217--gestalt.netlify.com

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please add a [changelog.md](https://github.com/pinterest/gestalt/blob/master/CHANGELOG.md) entry for your changes.
  </td>
    </tr>
  </tbody>
</table>



Filename | Size Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip
--- | --- | --- | --- | --- | --- | --- |
gestalt.js | :small_red_triangle_down: 0.8% | :small_red_triangle_down: 0.3% | 264.09 KB | 261.9 KB | 58.9 KB | 58.75 KB
gestalt.es.js | :small_red_triangle_down: 0.8% | :small_red_triangle_down: 0.3% | 261.54 KB | 259.35 KB | 58.54 KB | 58.38 KB
  
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please add a [changelog.md](https://github.com/pinterest/gestalt/blob/master/CHANGELOG.md) entry for your changes.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: packages/gestalt/src/Layer.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { createPortal } from 'react-dom';
+
+type Props = {|
+  children?: React.Node,
+|};
+
+export default class Layer extends React.Component<Props> {
+  constructor(props: Props) {
+    super(props);
+    this.el = document.createElement('div');
```
[the remark this answers is not in the document]
Comment (chrislloyd) [reply packages/gestalt/src/Layer.js]: Talking offline - I think the crashing behavior is actually the correct one here. However the error message will be fairly obtuse.

**Comment by christianvuerings** [comment]
@chrislloyd could we add some docs / example on how people can use this?

**Comment by chrislloyd** [comment]
@christianvuerings at the moment I'd like to make it internal only. I'm not sure if it's a great pattern / can be used well. I'd like to implement Toast/Modals with it first and see from there.

**Pull request merged**